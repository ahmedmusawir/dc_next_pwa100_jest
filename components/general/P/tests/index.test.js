import React from 'react';
import { mount, shallow } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import P from 'components/general/P';

// describe('<StyledComponent />', () => {
// it('should render a prop', () => {
// 	const id = 'testId';
// 	const wrapper = mount(<StyledComponent id={id} />);
// 	console.log(wrapper.debug());
// 	const renderedComponent = enzymeFind(wrapper, P);
// 	expect(renderedComponent.prop('id')).toEqual(id);
// });
// it('should render its text', () => {
// 	const children = 'Text';
// 	const wrapper = mount(<P>{children}</P>);
// 	const renderedComponent = enzymeFind(wrapper, P);
// 	expect(renderedComponent.contains(children)).toBe(true);
// });
// it('should render a text', () => {
// 	const wrapper = shallow(<P />);
// 	expect(wrapper.find('p').text()).toEqual('');
// });
// });

test('it works', () => {
	const tree = renderer.create(<P />).toJSON();
	expect(tree).toMatchSnapshot();
});

test('it works', () => {
	const wrapper = shallow(<P />);
	// console.log(wrapper.debug());
	expect(wrapper).toMatchSnapshot();
});
