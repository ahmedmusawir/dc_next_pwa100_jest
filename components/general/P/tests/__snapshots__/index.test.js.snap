// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.5rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
}

<p
  className="c0"
/>
`;

exports[`it works 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "forwardedComponent": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "P-sc-1m4acrm-0",
          "isStatic": false,
          "lastClassName": "kZUnVe",
          "rules": Array [
            "
    font-size: 1rem;
    line-height: 1.5rem;
    padding-top: 1rem;
    padding-bottom: 1rem;
",
            ";font-family:",
            [Function],
            ";color:",
            [Function],
            ";",
          ],
        },
        "displayName": "P",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "P-sc-1m4acrm-0",
        "target": "p",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "forwardedRef": null,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "forwardedComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "P-sc-1m4acrm-0",
            "isStatic": false,
            "lastClassName": "kZUnVe",
            "rules": Array [
              "
    font-size: 1rem;
    line-height: 1.5rem;
    padding-top: 1rem;
    padding-bottom: 1rem;
",
              ";font-family:",
              [Function],
              ";color:",
              [Function],
              ";",
            ],
          },
          "displayName": "P",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "P-sc-1m4acrm-0",
          "target": "p",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "forwardedRef": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
